### ISSUE TITLE
Item management + lvl requirements overhaul idea

### ISSUE THREAD

**Issue by Rychlas** [WarpzoneStudios/Hammerting#106]
I’ve summed up my idea for a temporary tool rework that can be done relatively fast and a longer one, that maybe in the future could be done. Please, everyone, share your thoughts about it as it for sure isn’t perfect and I really want to hear other ideas. Here we go:

Temporary tool rework:

tl;dr: Workshops have their own tool slot, user can choose a tool for the dwarf from the dwarf’s EQ, same for workshop. Details below:

1) Reserve 4 tool slots and make them tool-specific: 1 pickaxe slot, 1 weapon slot, 1 hammer slot, 1 utility slot (for armor, backpacks, climbing tools, etc) Only those tools can go into their respective slots. 
2) Add new, invisible for players temporary tool slot for dwarfs (needed later in point 7 as a hacky way to give dwarfs workshop tools. If it can be done in code simply by adding a code like “perform action with the bonuses of the workshop item” or “apply the equipped item bonuses to all recipes” or something then this is not needed).
3) Allow the players to search, choose and change a tool from the dwarf’s EQ UI. Upon left-clicking the tool slot a popup will appear. First, all available tool tiers (Copper, Iron, Mithril etc.) that are in the possession of the player (in any inventory).  Once the player clicks on a tier, a list will appear like when choosing a resource in workshop, with all tools/weapons of that tier that are in possession. Upon choosing, the dwarf will go to the inventory that has the item and equip it. 
4) Add a new tool slot for workshops. It should work the same way the one for the dwarfs works. LBM->Choose tier->Choose tool.
5) Create new tier 0 for any tool, called “Crude”, a default worst tier of tool that appears automatically when you build a workshop, as a starting tool with some penalties to speed/efficiency. 
6) Whenever a dwarf uses the workshop, COPY the current workshop  tool into the dwarf’s invisible temporary tool slot, then delete upon exiting the workshop. If it can be done in any other way allowing skipping the inventory interaction then great, this is just a hacky, roundabout suggestion.

Full tool rework – alright, buckle up:

Part I – Storage QoL + Item Request mechanics
1) Allow us to collapse inventories to show a list of items and their numbers instead in all inventories instead of a grid with all items (a checkbox per inventory [Item List] or [Collapse Inventory] or something). This will allow us to look through inventories with ease and allow us to perform new actions:
2) Add a slider with arrows and an input (you can slide, click and simply type in a number) to perform actions with a chosen number of items(sell, send, request):
2.a) Allow us to send items to chosen inventory (renaming buildings will come in handy).
2.b) Allow us to REQUEST items in any inventory, limited to items handled by that inventory (explained below)
3) Add a button to all inventories “Request”. It would open a UI that’d let you choose any item that could be used by that inventory (forge limited to ores, smithy limited to metals, wood, etc), to be requested. Upon requesting an item and its quantity, it will be delivered to the inventory.  if the item is NOT available, it will be greyed out but still available to be chosen – if chosen the game will automatically generate all required tasks in workshop to create the requested items. For example, we’d like to request 5 Iron Support Beams to our storage. Since we have only 2 Iron  Sheets, the request will automatically create crafting queues with high priority in respective workshops for 6 iron bars, 3 Iron Sheets in Foundry and for 5 Iron Support Beams in Smithy. Request button should also work when choosing a tool in a tool slot for dwarf or a workshop – An UI should pop up letting the player choose a tool tier and available to be crafted tool parts (if you can’t craft them because of locked knowledge or no one has required lvl to craft the item it will appear Yellow if you don’t have the level for it and RED if it is knowledge-locked instead of grey). When the item is crafted, it will remember that it was requested and ping back to be picked up and delivered or equipped to the requesting entity.

I I – Equipment slots
1) – Just like in temporary solution above, reserve 3 slots in dwarven inventory for 1 pickaxe, 1 weapon and 1 hammer. Implement the LMB tool choosing/requesting from the inventory of dwarf or workshop.
2) – Add 2-3 tool slots per Tier to Workshops separate from normal inventory. Allow workshops to choose 1 tool of any tier of any configuration in from their tool inventory. (It will work similar to unlocking various bonuses the players can then choose from whenever they want. Previous idea was to have it infinite, but 2-3 slots should be enough). Allow players to choose a preferred tool for each Tier of the tools if it has been unlocked. Dwarfs will use the best  tool tier they can (depending on their profession lvl), but of the type that the player choses. Add secondary box “enforce tool” with its own choice of any of the tools, that will force more experienced dwarfs to use a tool of player’s choice (for example for a specific tool bonus that’s not available in higher tiers).

 ![Workshop tool choice](https://i.snipboard.io/svk7Bf.jpg)

2) - Create a tool managing UI with two pages – One for dwarfs, one for workshops. 
it would be split in two, available tools on the left sorted by tier, with shown stats and quantity,
and workshops or dwarfs on the right, with currently equipped tools. If you press on a dwarf, better tools on the left will be highlighted and better stats will be shown in green. Pressing RMB on the item when a dwarf is selected will automatically equip the tool. Multiple dwarfs can be selected, and upon RMB clicking on a tool all of the selected dwarfs will equip it, with priority having those dwarfs whose level in that skill is the highest. If you RMB on the item without having any dwarf selected, you can choose to either sell chosen amount or equip all or chosen amount of that item to all dwarfs (skill priority applies if not enough tools)

![Tool managment - dwarfs](https://i.snipboard.io/uGdn7h.jpg)
![Tool management - workshops](https://i.snipboard.io/0pybct.jpg) 
 

III – Professions lvl requirements and Job Broker panel in Workshops

When the previous things are finished, the game would be ready to welcome a profession lvl requirements to use certain tools and craft things. So all ores, all craftable things, all tools would need to be revisited and would have to have their requirement lvls set.

Adding an ability to link a workshop to a selected dwarf or dwarfs would be very helpful.

When skilling up a dwarf, after setting his jobs in job broker and selecting him workplaces to work in, he’d go into the workshop and start working, skilling up, and using better and better tools available in the workshop based on his proficiency.

This would give much deeper meaning to professions lvls, and players would actually have to pay attention to job brokers.

IV – Additional utility tools, additional workshop tools and tool types

Each workshop would need a tool, and each tool should have head/handle types available for it to allow for various bonuses/penalties in crafting speed, energy usage, fuel usage etc.

I believe it would be also beneficial to add more utility tools that would perfectly fit into the new lvl requirements. Various backpacks depending on hauling lvl, climbing gear allowing very slow walking on walls, armor for some protection (or in a different armor slot, but having it in utility slot would kinda balance the game better I believe. No climbing/hauling militia!), lanterns to widen the spotting range of the dwarf, and many others. 

In a new slot (fifth) or in the existing utility slot, artifacts could be placed, looted from monster lairs, found in the corpses of other adventurers, won in overworld missions or crafted with the usage of the gems. 

V - Ability to craft artifacts
Barry__on discord suggested a great thing - ability to craft artifacts. In a new arcane workshop we could craft rune words from gems and then enchant tools with them. Depending on word tier, tool tier it'd require different Arcanist proffesion lvls and would destroy all materials upon failure.

So yeah, this is a rough sketch of my idea. Please, feel free to dismantle it, discuss and criticize.

[COPIED FROM DISCORD WITH A SMALL CHANGE]

**Comment by Srekel** [comment]
Super interesting, thanks :) I will read it through later but want to put down some thoughts I've had relating to this before breakfast. 

I've been in favor of tools in buildings for a good while, but, I'm not convinced yet. There are a few things that need to happen with the game before we should consider making that change.

For example, we want to retain the "RPG" feeling of equipping dwarves with cool stuff - you'd still want your expert blacksmith to have the best hammer. So maybe you could have a communal hammer, but the value of putting a better one on your blacksmith would be that they could form a bond with it, increasing its stats *with that dwarf* over time. 

One of the things I like about tools-in-buildings is that we could then have many more kinds of tools. Rakes make sense for certain Farm work but for other stuff, a spade or hoe might be better. It would be impossible to expect the farming dwarf to carry all of this with them, so putting them in the farm makes more sense. This would "ground" the game, make it more concrete, which is something I personally like.

It makes sense to have an epic hammer on a blacksmith, but it doesn't feel so epic to carry around an epic ladle :D But that is still something we would need to do, for consistency.

**Comment by Rychlas** [comment]
Yeah, what you're saying is very promising. I'd love for each dwarf to have a more rpg-style equipment, with different slots.
If you're really going into that direction, which is awesome, here's my little suggestion how it could look like.
I also suggest adding buttons to the character UI to split the info screen and to keep it to one window instead of opening a new one when opening talent tree for example (it takes up too much space!)

![EQ](https://i2.paste.pics/afe8ea6243960856b7f149581a49739d.png)
//https://i2.paste.pics/afe8ea6243960856b7f149581a49739d.png <- couldn't view img in preview so I'm pasting the link


Also, I believe your idea of "leveling" tools could be done with Runes. Upon engraving runes (or one of the various types) on a tool, it'd start gathering XP alongside the user. And I think it fits it being used in workshop too, Lore-wise, it'd be as if the working master craftsman is embedding a part of himself in the tool as he's working, so all his successors will feel his energy, his skill bestowed upon the tools he used. So that even if the dwarf dies, the player will know that a part of him is still in the mountainhome - in the tools he used which are his legacy now for the next generation of the dwarfs :)
